### ISSUE TITLE
Resolution modification upon `restart`

### ISSUE THREAD

**Issue by akshaysridhar** [CliMA/ClimateMachine.jl#1204]
# Description

Ability to `restart` simulations on a `coarser` / `finer` DG grid than the one used at the `checkpointing` stage. 

I include this here as a `wishlist` item (not currently prioritized), @szy21  @LenkaNovak @bischtob could you please comment on the utility of such a tool in the longer term? 

cc @kpamnany 

# For CLIMA Developers

- [ ] This feature can be added (if it cannot be, explain why in a comment below -- lack of technical expertise, not relevant to the scope of this project, too ambitious)
- [ ] There is a timeline for when this feature can be implemented
- [ ] The feature has been (or will be) implemented (Please link the PR)

**Comment by tapios** [comment]
Yes, this would be very useful. We have used such tools frequently to accelerate the spinup of simulations, before doing higher-resolution simulations. It would also be useful for the ocean (where spinup times are 1,000's of years).

**Comment by szy21** [comment]
It would be very useful for the LES too.
